clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -8913,6 +8915,9 @@ bool Sema::CheckExplicitlyDefaultedComparison(Scope *S, FunctionDecl *FD,
 
   // Perform any unqualified lookups we're going to need to default this
   // function.
+  if (!S)
```

Comment:
Based on @zygoloid comment:

https://github.com/llvm/llvm-project/issues/97087#issuecomment-2207341010

here it looks like the fix is the pass Scope instead of `nullptr` here:

https://github.com/llvm/llvm-project/blob/4c63672ca706c708de1e49bb29d026a705daa0d2/clang/lib/Sema/SemaDeclCXX.cpp#L17990

Did you try this?